src/components/uiOptions/js/UIOptions.js

```diff
@@ -53,4 +53,228 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
+        listeners: {
+            "{fluid.uiOptions.prefsEditor.localized}.events.onInterfaceLocaleChangeRequested": {
+                func: "{fluid.uiOptions.prefsEditor.localized}.reloadUioMessages",
+                args: ["{arguments}.0.data", "{that}.events.onLazyLoad"],
+                priority: "after:changeLocale",
+                namespace: "reloadUioMessages"
+            }
+        }
+    });
+
+    /* Reloads UIO message bundles with the given locale,
+     * subject to the fallback rules of the resourceLoader and,
+     * if provided, fires an event to trigger lazy loading
+     * - "locale": the locale for which to load messages
+     * - "uioMessageLoaderComponent": the UIO messageLoader component
+     * - "uioMessageLoaderLocalePath": the path on the messageLoader for its locale
+     * - "lazyLoadEvent": the event to fire to trigger lazy loading
+     */
+    fluid.uiOptions.prefsEditor.localized.reloadUioMessages = function (locale, uioMessageLoaderComponent, uioMessageLoaderLocalePath, lazyLoadEvent) {
+        // Set the locale in the resource loader
+        fluid.set(uioMessageLoaderComponent, uioMessageLoaderLocalePath, locale);
+
+        // Force the resource loader to get the new resources
+        fluid.resourceLoader.loadResources(uioMessageLoaderComponent, uioMessageLoaderComponent.resolveResources());
+
+        if (lazyLoadEvent) {
+            lazyLoadEvent.fire();
+        }
+    };
+
+    /* Updates a given localized component's messages with new values and sets a
+     * messageLocale value in order for the current locale to be verifiable
+     * - "localizedComponent": the component to be updated
+     * - "resourceKey": the key to access in the resources collection
+     *   where this component's messages are stored
+     * - "resources": a set of message bundles
+     * - "locale": the locale of the messages
+     */
+    fluid.uiOptions.prefsEditor.localized.updateLocalizedComponent = function (localizedComponent, resourceKey, resources, locale) {
+        if (localizedComponent.msgResolver) {
+            localizedComponent.msgResolver.messageLocale = locale;
+            localizedComponent.msgResolver.messageBase = resources[resourceKey].resourceText;
+        }
+    };
+
+    /* Updates and redraws the slidingPanel of a UIO component
+    * - "resources": a set of message bundles
+    * - "locale": the locale to be loaded/updated
+    * - "slidingPanel": the slidingPanel component to update
+    * - "slidingPanelStringMap": a map of msgLookup key names to bundle-specific key names
+    */
+    fluid.uiOptions.prefsEditor.localized.updateSlidingPanel = function (resources, locale, slidingPanel, slidingPanelStringMap) {
+        fluid.uiOptions.prefsEditor.localized.updateLocalizedComponent(slidingPanel, "prefsEditor", resources, locale);
+
+        slidingPanel.options.strings = fluid.transform(slidingPanel.options.strings, function (localizedString, key) {
```

Comment:
In upcoming versions of the framework, assignment to options will be impossible since they will be immutable. Rework this to use model fields instead (can find inspiration in the GPII app's messageBundles system - https://github.com/GPII/gpii-app/blob/master/src/shared/messageBundles.js#L98 )